python/simulation.py

```diff
@@ -3259,41 +3253,43 @@ def get_array_metadata(self, vol=None, center=None, size=None, dft_cell=None,
         $$ \\int_{\\mathcal V} Q(\\mathbf{x})d\\mathbf{x} \\approx \\sum_{n} w_n Q_n.$$
 
         This is a 1-, 2-, or 3-dimensional integral depending on the number of dimensions
-        in which $\\mathcal{V}$ has zero extent. If the $\\{Q_n\\}$ samples are stored in an
+        in which $\\mathcal{V}$ has zero extent. If the $Q_n$ samples are stored in an
         array `Q` of the same dimensions as `w`, then evaluating the sum on the RHS is
         just one line: `np.sum(w*Q).`
 
         A convenience parameter `dft_cell` is provided as an alternative to `vol` or
-        `center/size`; set `dft_cell` to a `dft_flux` or `dft_fields` object to define the
-        region covered by the array. If the `dft` argument is provided then all other
-        arguments (`vol`, `center`, and `size`) are ignored. If no arguments are provided,
-        then the entire cell is used.
+        `center`/`size`. Set `dft_cell` to a `dft_flux` or `dft_fields` object to define the
+        region covered by the array. If the `dft_cell` argument is provided then all other
+        arguments related to the spatial region (`vol`, `center`, and `size`) are ignored.
+        If no arguments are provided, then the entire cell is used.
+
+        For empty dimensions of the grid slice `get_array_metadata` will collapse
+        the *two* elements corresponding to the nearest Yee grid points into a *single*
+        element using linear interpolation.
+
+        If `return_pw=True`, the return value is a 2-tuple `(p,w)` where `p` (points) is a
+        list of `mp.Vector3`s with the same dimensions as `w` (weights). Otherwise, by
+        default the return value is a 4-tuple `(x,y,z,w)`.
         """
         if dft_cell:
-            vol, collapse = dft_cell.where, True
+            vol = dft_cell.where
         if vol is None and center is None and size is None:
             v = self.fields.total_volume()
         else:
             v = self._volume_from_kwargs(vol, center, size)
-        xyzw_vector = self.fields.get_array_metadata(v, collapse, snap)
+        xyzw_vector = self.fields.get_array_metadata(v)
         offset, tics = 0, []
         for n in range(3):
             N = int(xyzw_vector[offset])
             tics.append( xyzw_vector[offset+1:offset+1+N] )
             offset += 1+N
-        wshape=[len(t) for t in tics if len(t)>1]
-        weights=np.reshape(xyzw_vector[offset:],wshape)
+        wshape = [len(t) for t in tics if len(t)>1]
+        weights = np.reshape(xyzw_vector[offset:], wshape)
         if return_pw:
             points=[ mp.Vector3(x,y,z) for x in tics[0] for y in tics[1] for z in tics[2] ]
             return points,weights
         return tics + [weights]
```

Comment:
To return a tuple, do `return tuple(tics) + (weights,)`

Reply:
(We could also change `tics` earlier to be a tuple from the start, but this is the easiest 1-line change.)

Reply:
Fixed.